packer/scripts/windows/winrm_max_memory.ps1

```diff
@@ -0,0 +1,18 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+$memorygb = [int]$args[0]
+$memorygb
+
+if ($memorygb) {
+    $memorymb = $memorygb * 1024
+}
+else {
+    $memorymb = 30 * 1024
+}
+cmd /c winrm set "winrm/config/winrs" "@{MaxMemoryPerShellMB=\`"$memorymb\`"}"
+cmd /c winrm get winrm/config
```

Comment:
Can you please explain this? What it is trying to do?
Thanks!

Reply:
It is trying to get user input in GB and convert to MB and apply to WinRM max memory allocation during Jenkins agent node communications.

When Jenkins communicate with Windows node through WinRM it can only allocate the max number of memory in MBs that WinRM allows it to allocate.

This make sure the value is assigned dynamically based on the AMI requirements. Thanks.